server/shared/storage/amazon.ts

```diff
@@ -0,0 +1,123 @@
+import IStorage, {
+  ContentResponse,
+  DeleteResponse,
+  FileListResponse
+} from './interface';
+import { AWSError } from 'aws-sdk';
+import { Config } from '../../config';
+import miss from 'mississippi';
+import path from 'path';
+import S3 from 'aws-sdk/clients/s3';
+
+const isNotFound = (err: AWSError): boolean => err.code === 'NoSuchKey';
+
+class Amazon implements IStorage {
+  #bucket: string;
+  #client: S3;
+
+  constructor(config: Config) {
+    const amazonConfig = config.storage.amazon;
+
+    const s3Config = {
+      accessKeyId: amazonConfig.key,
+      secretAccessKey: amazonConfig.secret,
+      region: amazonConfig.region,
+      signatureVersion: 'v4',
+      apiVersion: '2006-03-01',
+      maxRetries: 3
+    };
+
+    this.#bucket = amazonConfig.bucket;
+    this.#client = new S3(s3Config);
+  }
+
+  // API docs: http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#getObject-property
```

Comment:
Do we need API docs comment for every storage method? If docs are useful can we just link the main docs page instead of linking docs for every method we use (some of them even twice)?

Reply:
I definitely agree with the statement that some of them that are repeating, but still it can be handy when somebody wants to change the implementation of the method. @MiroDojkic thoughts?

Reply:
@malizeus I think these are handy, but I also understand Ivo's point. I'm ok with both a single link or link for each method.

Reply:
@MiroDojkic also we should consider bumping the version to v3 in the future since it's stable and recommended for general usage (it's written in v2 docs 😄 ).